### PR TITLE
[Tests] Limit layernorm CK test applicability

### PR DESCRIPTION
**Pull request by junliume**
#2527

### CODE REVIEW

File: test/gtest/layernorm.cpp [hunk 1]
```diff
@@ -41,7 +41,11 @@ struct LayerNormTestFloat : LayerNormTest<float>
 
 TEST_P(LayerNormTestFloat, LayerNormTestFw)
 {
-    if(!(miopen::IsEnvvarValueEnabled("MIOPEN_TEST_ALL")) && (GetFloatArg() != "--float"))
+    const auto& handle = get_handle();
+    if(!(miopen::IsEnvvarValueEnabled("MIOPEN_TEST_ALL")) && (GetFloatArg() != "--float") ||
```
Comment (atamazov) [inline test/gtest/layernorm.cpp]: [R] 
```suggestion
    if(!(miopen::IsEnabled("MIOPEN_TEST_ALL")) && (GetFloatArg() != "--float") ||
```

**Review by atamazov — APPROVED** [review]
LGTM

**Review by iq136boy — APPROVED** [review]

**Review by seungmanhan — APPROVED** [review]
Sorry, I didn't know that the composable kernel was only supported in MI100, MI200 and MI300.
LGTM.

**Comment by junliume** [comment]
@seungmanhan please re-review the latest update, the original non-applicable condition does not look correct to me:
```
if(!(miopen::IsEnvvarValueEnabled("MIOPEN_TEST_ALL")) && (GetFloatArg() != "--float"))
```
maybe it should be
```
if(!(miopen::IsEnvvarValueEnabled("MIOPEN_TEST_ALL") && (GetFloatArg() != "--float")))
```

**Comment by seungmanhan** [comment]
> miopen::IsEnvvarValueEnabled("MIOPEN_TEST_ALL")

only "!(miopen::IsEnvvarValueEnabled("MIOPEN_TEST_ALL"))" is true and "(GetFloatArg() != "--float")" is ture, then skip
If either one is false, we do not skip.
"!(miopen::IsEnvvarValueEnabled("MIOPEN_TEST_ALL"))" is false mean test all
"(GetFloatArg() != "--float")" is false mean test float
So code shuld be below. I missed it.
```
    if((!miopen::IsEnvvarValueEnabled("MIOPEN_TEST_ALL") && (GetFloatArg() != "--float")) ||
       !(miopen::StartsWith(handle.GetDeviceName(), "gfx908") ||
         miopen::StartsWith(handle.GetDeviceName(), "gfx90a") ||
         miopen::StartsWith(handle.GetDeviceName(), "gfx94")))
```
and (GetFloatArg() != "--float") -> !(GetFloatArg() == "--float") looks better.

**Comment by junliume** [comment]
Thanks, @seungmanhan !

@JehandadKhan  and @xinlipn , then it seems that the following two locations have logic issues:

- Should the test be enabled when `test_all` **OR** `test_float/half`??

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/60cf9c095356a58d9ef369856b585ac1f4d39946/test/gtest/conv_igemm_mlir.cpp#L117-L120

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/60cf9c095356a58d9ef369856b585ac1f4d39946/test/gtest/conv_igemm_mlir.cpp#L139-L142

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/60cf9c095356a58d9ef369856b585ac1f4d39946/test/gtest/conv_igemm_mlir_xdlops.cpp#L90-L93

CC: @atamazov

**Comment by seungmanhan** [comment]
@junliume 

https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2320#discussion_r1327817140
I misunderstood about MIOPEN_TEST_ALL. MIOPEN_TEST_ALL is a value unrelated to type.
So below is right.
I am sorry for the repeated edits

```
    if(!(miopen::IsEnvvarValueEnabled("MIOPEN_TEST_ALL") && (GetFloatArg() == "--float")) ||
       !(miopen::StartsWith(handle.GetDeviceName(), "gfx908") ||
         miopen::StartsWith(handle.GetDeviceName(), "gfx90a") ||
         miopen::StartsWith(handle.GetDeviceName(), "gfx94")))
```

**Comment by atamazov** [comment]
@seungmanhan 
> I misunderstood about MIOPEN_TEST_ALL. MIOPEN_TEST_ALL is a value unrelated to type.

Yes. These are old foggy things that we never had time to clear.

File: test/gtest/layernorm.cpp [hunk 2]
```diff
@@ -41,7 +41,10 @@ struct LayerNormTestFloat : LayerNormTest<float>
 
 TEST_P(LayerNormTestFloat, LayerNormTestFw)
 {
-    if(!(miopen::IsEnvvarValueEnabled("MIOPEN_TEST_ALL")) && (GetFloatArg() != "--float"))
+    if(!(miopen::IsEnvvarValueEnabled("MIOPEN_TEST_ALL") && (GetFloatArg() == "--float")) ||
+       !(miopen::StartsWith(handle.GetDeviceName(), "gfx908") ||
+         miopen::StartsWith(handle.GetDeviceName(), "gfx90a") ||
+         miopen::StartsWith(handle.GetDeviceName(), "gfx94")))
```
Comment (atamazov) [inline test/gtest/layernorm.cpp:48]: [Note] Here's how to remove extra parentheses and simplify the expression.
```suggestion
    if(!(miopen::IsEnvvarValueEnabled("MIOPEN_TEST_ALL")          //
        && GetFloatArg() == "--float"                             //
        && (miopen::StartsWith(handle.GetDeviceName(), "gfx908")  //
          || miopen::StartsWith(handle.GetDeviceName(), "gfx90a") //
          || miopen::StartsWith(handle.GetDeviceName(), "gfx94"))))
```

**Pull request merged**